packages/core-data/src/utils/crdt-blocks.ts

```diff
@@ -187,3 +195,37 @@ export function mergeBlocks(
 		knownClientIds.add( clientId );
 	}
 }
+
+/**
+ * Determine if a block should be synced.
+ *
+ * Ex: A gallery block should not be synced until the images have been
+ * uploaded to WordPress, and their url is available. Before that,
+ * it's not possible to access the blobs on a client as those are
+ * local.
+ *
+ * @param block The block to check.
+ * @return True if the block should be synced, false otherwise.
+ */
+function shouldBlockBeSynced( block: Block ): boolean {
+	// Verify that the gallery block is ready to be synced.
+	// This means that, all images have had their blobs converted to full URLs.
+	// Checking for only the blobs ensures that blocks that have just been inserted work as well.
+	if ( 'core/gallery' === block.name ) {
+		return ! block.innerBlocks.some(
+			( innerBlock ) =>
+				innerBlock.attributes && innerBlock.attributes.blob
+		);
+	}
+
+	// Except the gallery block, all the other core blocks should be synced.
+	// We don't want anyone to override that, as that'll cause problems.
+	if ( block.name.startsWith( 'core/' ) ) {
```

Comment:
@ingeniumed For non-core blocks, what happens when `shouldBlockBeSynced()` is false? Does only 1 user get the block?

Reply:
The idea would be that, none of the ydocs would have the block including the user who made the edit in Gutenberg. The block is considered to be "not ready to be synced" so it would be held off from being sent to all the ydocs. Presumably, whoever is using this filter would use it similar to the gallery block where after a certain condition is met it would be flipped to true and allowed to sync.

There'd def be a weird flicker where the block appears, vanishes as it's made "ready" and then shows up once its ready. But, it's better than the way the gallery block works now where you need to refresh the page.

All that said, I figured this would allow us to provide support for plugins that are out of our control or plugins like our own - RDB in their own way outside of gutenberg.